### PR TITLE
[PBW-3807] centering video in iPad

### PR DESCRIPTION
**Pull request by ksunika**

### CODE REVIEW

File: js/styles/inlineStyle.js [hunk 1]
```diff
@@ -11,6 +11,24 @@ module.exports = {
     }
   },
 
+  iPadVerticalVideoStyle: {
```
Comment (mistermendez) [inline js/styles/inlineStyle.js]: is there a way to do this with CSS instead of inline JS style?

File: js/controller.js [hunk 2]
```diff
@@ -146,6 +151,9 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       $("#" + elementId + " .player_skin").css("z-index", OO.CSS.ALICE_SKIN_Z_INDEX);
       this.state.mainVideoElement = $("#" + elementId + " .video");
 
+      this.state.playerHeight = $("#" + elementId + " .innerWrapper").height();
```
Comment (mistermendez) [inline js/controller.js]: Can we do this without direct DOM manipulation (jQuery) and stick to React's virtual DOM?

It's considered best practice to not mix direct DOM manipulation with React.

See this for more details - https://facebook.github.io/react/docs/working-with-the-browser.html

Comment (ksunika) [inline js/controller.js]: We should probably then change the above statements from jQuery to React as well. I am not sure about how to do it right now.
@chibzu, your thoughts?

File: js/styles/inlineStyle.js (hunk 1 above)
Comment (ksunika) [inline js/styles/inlineStyle.js]: I am not sure that using the changing class name as a way to apply a different style for the main video element is completely safe. 
@chibzu, what do you think?

Comment (aldigun) [inline js/styles/inlineStyle.js]: I would think that the 4 bottom can be put on CSS. While width and height can be assigned on the fly in controller.js

File: js/controller.js (hunk 2 above)
Comment (aldigun) [inline js/controller.js]: AFAIK, innerWrapper is not made of react, so it is fine.
player_skin is the one we need to be careful of not mixing.

File: js/controller.js [hunk 3]
```diff
@@ -45,6 +45,9 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       "elementId": null,
       "buffering": false,
       "mainVideoPlayhead": null,
+      "playerHeight": null,
+      "playerWidth": null,
+      "iPadVideoPositionSet": false,
```
Comment (aldigun) [inline js/controller.js]: iPadVideoPositionSet is a weird naming

File: js/controller.js [hunk 4]
```diff
@@ -92,6 +95,8 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       this.mb.subscribe(OO.EVENTS.CONTENT_TREE_FETCHED, 'customerUi', _.bind(this.onContentTreeFetched, this));
       this.mb.subscribe(OO.EVENTS.AUTHORIZATION_FETCHED, 'customerUi', _.bind(this.onAuthorizationFetched, this));
       this.mb.subscribe(OO.EVENTS.PLAYING, 'customerUi', _.bind(this.onPlaying, this));
+      this.mb.subscribe(OO.EVENTS.VC_VIDEO_ELEMENT_IN_FOCUS, 'customerUi', _.bind(this.onVideoControllerVideoElementInFocus, this));
```
Comment (aldigun) [inline js/controller.js]: Do we really need to rely on in_focus and lost_focus event? Seems an overkill

File: js/controller.js (hunk 3 above)
Comment (ksunika) [inline js/controller.js]: how about iPadVideoCentered?

File: js/controller.js [hunk 5]
```diff
@@ -262,8 +270,40 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
...
+    onVideoControllerVideoElementInFocus: function() {
+      if (Utils.isIPad() && this.state.iPadVideoPositionSet == false) {
+        //hiding the video before it is positioned correctly by setIPadVideoPosition
+        this.state.mainVideoElement.css({"visibility":"hidden"});
+      }
+    },
+
     onPlaying: function() {
       // pause/resume of Ad playback is handled by different events => WILL_PAUSE_ADS/WILL_RESUME_ADS
+      if (Utils.isIPad() && this.state.iPadVideoPositionSet == false) {
```
Comment (aldigun) [inline js/controller.js]: Couple comments:
1. this.state.iPadVideoPositionSet == false inside IF condition can be refactored to !this.state.iPadVideoPositionSet
2. I would recommend to move isIpad and iPadVideoPositionSet check inside the setIPadVideoPosition()

File: js/controller.js (hunk 4 above)
Comment (ksunika) [inline js/controller.js]: in in_focus we make the main video hidden, and then show it later. Without it, we have the video jumping from the side to the center and it is very visible.

File: js/controller.js [hunk 6]
```diff
@@ -262,8 +270,40 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       }
     },
 
+    setIPadVideoPosition: function() {
```
Comment (aldigun) [inline js/controller.js]: - weird naming as well. I suggest setVideoCentering?
- Put a comment explaining this method.

**Comment by aldigun** [comment]
After discussion, this fix should be done by HTML5 video plugin rather than skin.
Will port the fix in the correct side and closing this merge.

**Pull request closed without merging**